### ISSUE TITLE
ThirstHealthFix Saturation Bug

### ISSUE THREAD

**Issue by dannydjdk** [p455w0rd/ToughExpansion#35]
When ThirstHealthFix option is enabled, all food saturation is drained whenever player is not fully hydrated.

See https://github.com/Glitchfiend/ToughAsNails/issues/527
per kd8lvt
> In the ToughExpansion config there is a option called "ThirstHealthFix" which requires your thirst to be full for you to heal. However, all it does is the block healing event itself (thus preventing you from refilling hearts), rather than saying "no, you can't heal because you don't have the food" (aka the proper way of doing it) which confuses Minecraft, resulting in your saturation to go down (or the healing potion to be drunk, etc.) but not heal you.

>Disabling this config option (which is enabled by default) fixes this problem.

I've experienced this bug with the following:
Tough Expansion 3.4.24
Forge: 14.23.3.2655
MC: 1.12.2

**Comment by SatyrRex** [comment]
Can confirm, also happens to me. Also, healing potions and golden apples do not work. When using the mod Rough Tweaks, which also disables health regeneration, none of the first aid kits work either. With the mod Dissolution, fairies are supposed to give you regeneration, and while you get the buff, you don't get healed. 

TE: 3.4.24
Forge: 14.23.4.2739
MC: 1.12.2

**Comment by Pyrolusite** [comment]
As of today, this bug still exists and negates any health regeneration effect as long as thirst is equal or below 14 points.

**Comment by dannydjdk** [comment]
To work around this bug, you can disable ThirstHealthFix in the config. Perhaps the mod author could have it disabled by default if this bug isn't going to be fixed soon.

**Comment by Pyrolusite** [comment]
Yeah, that's what I had to do. That's a shame, I liked that feature, but it's way too punitive atm.
